examples/homography_estimation.py

```diff
@@ -0,0 +1,463 @@
+    anim_dir = f"{log_dir}/animation"
+    os.makedirs(anim_dir, exist_ok=True)
+    subsample_anim = 1
+    H8_1_2_hist = H_hist["H8_1_2"]
+    num_iters = H8_1_2_hist.shape[-1]
+    for it in range(num_iters):
+        if it % subsample_anim != 0:
+            continue
+        # Visualize only first element in batch.
+        H8_1_2 = H8_1_2_hist[..., it]
+        H_1_2 = torch.cat([H8_1_2, H8_1_2.new_ones(H8_1_2.shape[0], 1)], dim=-1).to(
+            Hgt_1_2.device
+        )
+        H_1_2_mat = H_1_2[0].reshape(1, 3, 3)
+        Hgt_1_2_mat = Hgt_1_2[0].reshape(1, 3, 3)
+        imgH, imgW = img1.shape[-2], img1.shape[-1]
+        fc_err = four_corner_dist(H_1_2_mat, Hgt_1_2_mat, imgH, imgW)
+        if err_hist is None:
+            err = -1
+        else:
+            err = float(err_hist[0][it])
+        img1 = img1[0][None, ...]
+        img2 = img2[0][None, ...]
+        img1_dsts = warp_perspective_norm(H_1_2_mat, img1)
+        path = os.path.join(log_dir, f"{anim_dir}/{it:05d}.png")
+        viz_warp(path, img1[0], img2[0], img1_dsts[0], it, err=err, fc_err=fc_err)
+    anim_path = os.path.join(log_dir, "animation.gif")
+    cmd = f"convert -delay 10 -loop 0 {anim_dir}/*.png {anim_path}"
+    print("Generating gif here: %s" % anim_path)
+    os.system(cmd)
+    shutil.rmtree(anim_dir)
+    return
+
+
+def four_corner_dist(H_1_2, H_1_2_gt, height, width):
+    Hinv_gt = torch.inverse(H_1_2_gt)
+    Hinv = torch.inverse(H_1_2)
+    grid = kornia.utils.create_meshgrid(2, 2, device=Hinv.device)
+    warped_grid = kornia.geometry.transform.homography_warper.warp_grid(grid, Hinv)
+    warped_grid_gt = kornia.geometry.transform.homography_warper.warp_grid(
+        grid, Hinv_gt
+    )
+    warped_grid = (warped_grid + 1) / 2
+    warped_grid_gt = (warped_grid_gt + 1) / 2
+    warped_grid[..., 0] *= width
+    warped_grid[..., 1] *= height
+    warped_grid_gt[..., 0] *= width
+    warped_grid_gt[..., 1] *= height
+    dist = torch.norm(warped_grid - warped_grid_gt, p=2, dim=-1)
+    dist = dist.mean(dim=-1).mean(dim=-1)
+    return dist
+
+
+class SimpleCNN(nn.Module):
+    def __init__(self, D=32):
+        super(SimpleCNN, self).__init__()
+        self.relu = nn.ReLU(inplace=True)
+        self.conv1 = nn.Conv2d(3, D, kernel_size=3, stride=1, padding=1)
+        self.conv2 = nn.Conv2d(D, 3, kernel_size=3, stride=1, padding=1)
+        self.bn1 = nn.BatchNorm2d(D)
+
+    def forward(self, img):
+        x = self.relu(self.bn1(self.conv1(img)))
+        return self.conv2(x)
+
+
+def run():
+    batch_size = 64
+    max_iterations = 50
+    step_size = 0.1
+    verbose = True
+    imgH, imgW = 60, 80
+    use_gpu = True
+    viz_every = 10
+    disp_every = 10
+    save_every = 100
+    num_epochs = 999
+    outer_lr = 1e-4
+    use_cnn = True
+
+    log_dir = os.path.join(os.getcwd(), "viz")
+    os.makedirs(log_dir, exist_ok=True)
+    print("Writing to tensorboard at", os.getcwd())
```

Comment:
I don't think we need the tensorboard dependency, we can just log the loss to a csv file or something similar.

Reply:
@ddetone @joeaortiz

Reply:
Sounds good to me, replacing tensorboard with writing to a csv

Reply:
I removed the logging of the loss altogether as it is being printed anyway.